ui/summary.go

```diff
@@ -31,99 +30,119 @@ import (
 
 	"github.com/loadimpact/k6/lib"
 	"github.com/loadimpact/k6/stats"
+	"github.com/pkg/errors"
 	"golang.org/x/text/unicode/norm"
 )
 
 const (
-	GroupPrefix   = "█"
-	DetailsPrefix = "↳"
+	groupPrefix   = "█"
+	detailsPrefix = "↳"
 
-	SuccMark = "✓"
-	FailMark = "✗"
+	succMark = "✓"
+	failMark = "✗"
 )
 
 var (
-	ErrStatEmptyString            = errors.New("invalid stat, empty string")
-	ErrStatUnknownFormat          = errors.New("invalid stat, unknown format")
-	ErrPercentileStatInvalidValue = errors.New("invalid percentile stat value, accepts a number")
+	errStatEmptyString            = errors.New("invalid stat, empty string")
+	errStatUnknownFormat          = errors.New("invalid stat, unknown format")
+	errPercentileStatInvalidValue = errors.New("invalid percentile stat value, accepts a number")
 )
 
-var TrendColumns = []TrendColumn{
-	{"avg", func(s *stats.TrendSink) float64 { return s.Avg }},
-	{"min", func(s *stats.TrendSink) float64 { return s.Min }},
-	{"med", func(s *stats.TrendSink) float64 { return s.Med }},
-	{"max", func(s *stats.TrendSink) float64 { return s.Max }},
-	{"p(90)", func(s *stats.TrendSink) float64 { return s.P(0.90) }},
-	{"p(95)", func(s *stats.TrendSink) float64 { return s.P(0.95) }},
+// ErrInvalidStat represents an invalid trend column stat
+type ErrInvalidStat struct {
+	name string
+	err  error
 }
 
-type TrendColumn struct {
-	Key string
-	Get func(s *stats.TrendSink) float64
+func (e ErrInvalidStat) Error() string {
+	return errors.Wrapf(e.err, "stat '%s'", e.name).Error()
 }
 
-// VerifyTrendColumnStat checks if stat is a valid trend column
-func VerifyTrendColumnStat(stat string) error {
-	if stat == "" {
-		return ErrStatEmptyString
+// Summary handles test summary output
+type Summary struct {
+	trendColumns        []string
+	trendValueResolvers map[string]func(s *stats.TrendSink) interface{}
```

Comment:
Naming suggestions welcome here...

Technically, the `map` functions aren't columns (to show), but functions which _resolve_ the metric value, which is then mapped to a column, so I thought the naming distinction was as important as not keeping track of both things with one data structure.

Reply:
This name LGTM